dkist/net/helpers.py

```diff
@@ -70,7 +70,10 @@ def transfer_complete_datasets(datasets: Union[str, QueryResponseRow, DKISTQuery
     from dkist.net import conf
 
     if isinstance(datasets, str):
-        datasets = _get_dataset_inventory(datasets)[0]
+        try:
+            datasets = _get_dataset_inventory(datasets)[0]
+        except IndexError as e:
+            raise IndexError(f"No results available for dataset {datasets}") from e
```

Comment:
If we do this inside `_get_dataset_inventory` we could avoid the `try` block which is always my slight preference?